docs/specs/authentication-flow.md

```diff
@@ -841,3 +842,104 @@ Supertokens requires the developer to host a backend server to interactive with
 ### JSON schema
 
 Refer to the source code.
+
+### Action Data
+
+This section lists all possible types of data of `result.action.data`.
+
+Developer could identify the data type by checking the type key in `result.action.data.type`. All possible values are listed below:
+
+#### identification_data
+
+The data contains identification options.
+
+`options`: The list of usable identification options.
+
+#### authentication_data
+
+The data contains authentication options.
+
+`options`: The list of usable authentication options.
+
+#### oauth_data
+
+The data contains information for initiating an oauth authentication.
+
+`alias`: The configured alias of the selected oauth provider.
+`oauth_provider_type`: The type of the oauth provider, such as `google`.
+`oauth_authorization_url`: The authorization url of the oauth provider.
+`wechat_app_type`: The wechat app type. Only used when provider is `wechat`.
+
+#### create_authenticator_data
+
+The data contains options for creating new authenticator.
+
+`options`: The list of creatable authenticators.
+
+#### view_recovery_code_data
+
+The data contains recovery codes of the user.
+
+`recovery_codes`: The recovery codes of the user.
+
+#### oob_channels_data
```

Comment:
Rename this to `select_oob_otp_channels_data`

Reply:
Updated, thanks!